.github/workflows/main.yml

```diff
@@ -35,40 +33,49 @@ jobs:
         uses: actions/upload-artifact@v2
         with:
           name: build-tests-linux
-          path: cpp/build/tests
+          path: cpp/build
```

Comment:
This was the reason, why the code coverage for cpp did not work. We need to artifact the whole build folder for that, not just the tests subdirectory. This is a bit of a pain, because it is quite big, but it is a working solution. I propose to only try to fix this if we are too annoyed by this CI step taking to long. IMHO this should not be a blocker for this PR.

Reply:
OK, so we keep in mind to track the time needed for the CI from time to time :)